ghproxy/ghmetrics/ghpaths.go

```diff
@@ -0,0 +1,210 @@
+	case "statuses":
+		return fmt.Sprintf("%s%s%s", sanitizedPath, "/statuses", handleConstantAndVar(rest))
+	case "subscribers":
+		return fmt.Sprintf("%s%s%s", sanitizedPath, "/subscribers", handleConstantAndVar(rest))
+	case "branches":
+		return fmt.Sprintf("%s%s", sanitizedPath, handlePrefixedVarAndConstant("/branches", rest))
+	case "assignees":
+		return fmt.Sprintf("%s%s%s", sanitizedPath, "/assignees", handleConstantAndVar(rest))
+	case "git":
+		return fmt.Sprintf("%s%s", sanitizedPath, handlePrefixedConstantAndVar("/git", rest))
+
+	case "archive": // special path
+		return fmt.Sprintf("%s%s%s", sanitizedPath, "/archive", handleConstantAndVar(rest))
+
+	case "merges", "stargazers", "notifications", "hooks":
+		return fmt.Sprintf("%s%s", sanitizedPath, rest)
+	default:
+		logrus.WithField("sanitizedPath", sanitizedPath).WithField("rest", rest).Info("Path not handled")
+		return fmt.Sprintf("%s%s", sanitizedPath, rest)
+	}
+}
+
+func handlePrefixedVarAndConstant(prefix, path string) string {
+	pathTrimmed := strings.Replace(path, prefix, "", 1)
+	fragment := getFirstFragment(pathTrimmed)
+	switch fragment {
+	case "comments":
+		return fmt.Sprintf("%s/%s%s", prefix, fragment, handleConstantAndVar(pathTrimmed))
+	case "events":
+		return fmt.Sprintf("%s/%s%s", prefix, fragment, handleConstantAndVar(pathTrimmed))
+	}
+
+	match := varAndConstantRegex.FindStringSubmatch(pathTrimmed)
+	result := make(map[string]string)
+	for i, name := range varAndConstantRegex.SubexpNames() {
+		if i != 0 && name != "" && i <= len(match) { // skip first and empty
+			if name == "var" && match[i] != "" {
+				result[name] = ":var" // mask issue number
+			} else {
+				result[name] = match[i]
+			}
+		}
+	}
+	rest := result["path"]
+	sanitizedPath := fmt.Sprintf("%s/%s", prefix, result["var"])
+	if result["var"] == "" && rest == "" {
+		return prefix
+	} else if rest == "" || rest == "/" {
+		return sanitizedPath
+	}
+	return fmt.Sprintf("%s/%s%s", prefix, result["var"], rest)
+}
+
+func handlePrefixedConstantAndVar(prefix, path string) string {
+	pathTrimmed := strings.Replace(path, prefix, "", 1)
+	fragment := getFirstFragment(pathTrimmed)
+	if fragment != "" {
+		return fmt.Sprintf("%s/%s%s", prefix, fragment, handleConstantAndVar(pathTrimmed))
+	}
+
+	match := varAndConstantRegex.FindStringSubmatch(pathTrimmed)
+	result := make(map[string]string)
+	for i, name := range varAndConstantRegex.SubexpNames() {
+		if i != 0 && name != "" && i <= len(match) { // skip first and empty
+			if name == "var" && match[i] != "" {
+				result[name] = ":var" // mask issue number
+			} else {
+				result[name] = match[i]
+			}
+		}
+	}
+	rest := result["path"]
+	sanitizedPath := fmt.Sprintf("%s/%s", prefix, result["var"])
+	if result["var"] == "" && rest == "" {
+		return prefix
+	} else if rest == "" || rest == "/" {
+		return sanitizedPath
+	}
+	return fmt.Sprintf("%s/%s%s", prefix, result["var"], rest)
+}
+
+func handleConstantAndVar(path string) string {
```

Comment:
I'm not certain regex was the simplest choice here. If I understand it correctly, this method takes a string like:

```
/constant/var
```

Then turns it into

```
/constant/:var
```

Couldn't we implement it with:

```go
func handleConstantAndVar(path string) (string, error) {
    parts := strings.Split(path, "/")
    if len(parts) != 2 {
        return fmt.Errorf("could not handle path as a constant and variable: %q", path)
    }
    return fmt.Sprintf("/%s/:var", parts[0])
}
```

Reply:
Function is also handling the case, when there's no var attached, then it only returns the constant without the trailing slash, so I'll have to go back and change a little more, but can do, yes.